Documentation/gettingstarted/clustermesh.rst

```diff
@@ -359,54 +359,172 @@ the ``--cluster-name`` agent option or ``cluster-name`` ConfigMap option.
+      certificates to connect to that etcd.
+ 
+    * Run a ``kubectl exec -ti [...] bash`` in one of the Cilium pods and check
+      the contents of the directory ``/var/lib/cilium/clustermesh/``. It must
+      contain a configuration file for each remote cluster along with all the
+      required SSL certificates and keys. The filenames must match the cluster
+      names as provided by the ``--cluster-name`` argument or ``cluster-name``
+      ConfigMap option. If the directory is empty or incomplete, regenerate the
+      secret again and ensure that the secret is correctly mounted into the
+      DaemonSet.
 
- * When using global services, ensure that the ``cilium-operator`` deployment
-   is running and healthy. It is responsible to propagate Kubernetes services
-   into the kvstore. You can validate the correct functionality of the operator
-   by running:
+ #. Validate that the connection to the remote cluster could be established.
+    You will see a log message like this for each remote cluster:
 
-   .. code:: bash
+    .. code:: bash
+
+       level=info msg="Connection to remote cluster established"
+
+    If the connection failed, you will see a warning like this:
+
+    .. code:: bash
+
+       level=warning msg="Unable to establish etcd connection to remote cluster"
+
+    If the connection fails, the cause can be one of the following:
+
+    * Validate that the ``hostAliases`` section in the Cilium DaemonSet maps
+      each remote cluster to the IP of the LoadBalancer that makes the remote
+      control plane available.
+
+    * Validate that a local node in the source cluster can reach the IP
+      specified in the ``hostAliases`` section. The ``clustermesh-secrets``
+      secret contains a configuration file for each remote cluster, it will
+      point to a logical name representing the remote cluster:
+
+      .. code:: yaml
+
+         endpoints:
+         - https://cluster1.mesh.cilium.io:2379
+
+      The name will *NOT* be resolvable via DNS outside of the cilium pod. The
+      name is mapped to an IP using ``hostAliases``. Run ``kubectl -n
+      kube-system get ds cilium -o yaml`` and grep for the FQDN to retrieve the
+      IP that is configured. Then use ``curl`` to validate that the port is
+      reachable.
+
+    * A firewall between the local cluster and the remote cluster may drop the
+      control plane connection. Ensure that port 2379/TCP is allowed.
+
+State Propagation
+=================
+
+ #. Run ``cilium node list`` in one of the Cilium pods and validate that it
+    lists both local nodes and nodes from remote clusters. If this discovery
+    does not work, validate the following:
+
+    * In each cluster, check that the kvstore contains information about
+      *local* nodes by running:
+
+      .. code:: bash
+
+          cilium kvstore get --recursive cilium/state/nodes/v1/
+
+      .. note::
+
+	 The kvstore will only contain nodes of the **local cluster**. It will
```

Comment:
nit: The tabbing seems a bit weird to me in github interface. As long as the indentation looks right then that's fine.